release_bot/cli.py

```diff
@@ -18,30 +18,48 @@
 from pathlib import Path
 
 from release_bot.configuration import configuration
+from release_bot.exceptions import ReleaseException
 
 
 class CLI:
+
     @staticmethod
     def parse_arguments():
         """Parse application arguments"""
+
+        def run_init(_):
+            """
+            Makes the bot to run the init script when the `release_bot init`
+            """
+            configuration.init = True
+
+        def run_bot(args):
+            """
+            Makes the bot to perform normal release tasks
+            """
+            if args.configuration:
+                args.configuration = Path(args.configuration).resolve()
+                if not args.configuration.is_file():
+                    raise ReleaseException(
+                        f"Supplied configuration file is not found: {args.configuration}")
+                    if args.debug:
+                        configuration.logger.setLevel(logging.DEBUG)
+                        for key, value in vars(args).items():
```

Comment:
This should not be in the `if debug` branch.

Reply:
> I understand that argparse is hard to work with, please see my patch above and a link how you can easily do subcommands with argparse.

Hey did you remove the patch?? When I first saw your review there was some code edits ?

Reply:
@TomasTomecek In the cli currently I first add the configuration.init

> I just tried this and here is my observation:
> 
> * Why's gitchangelog in the secret config file and not in the release-conf?
> * The prompting part is inconsistent (some lines start with a big letter, some don't)
> * The prompting part is hard to read - a wall of text: I would much prefer if the init command created the files for me and then wrote instructions how to update them instead of prompting me: I hate prompts. I guess this could configurable: default to the interactive prompt (which is better for new users) and offer a non-interactive run.
> * Then I was finally able to run release bot and this was the result: [packit-service/ogr#38](https://github.com/packit-service/ogr/pull/38), no idea why the changelog is empty: I didn't find any clues in the debug output - no idea what went wrong.
> 
> Edit: figured it out: mustache was not installed, I have a changelog now: [packit-service/ogr#39](https://github.com/packit-service/ogr/pull/39)
> Edit2: The changelog generated by gitchangelog is better than what we had but still needs some tinkering (the template mainly), we can do that in next PRs.
> 
> And finally, I had to signifiicantly patch release-bot to make it run - the CLI code seems to be pretty borked.
> 
> ```diff
> diff --git a/release_bot/cli.py b/release_bot/cli.py
> index e18e32c..be21d12 100644
> --- a/release_bot/cli.py
> +++ b/release_bot/cli.py
> @@ -37,15 +37,6 @@ class CLI:
>              """
>              Makes the bot to perform normal release tasks
>              """
> -            if args.configuration:
> -                args.configuration = Path(args.configuration).resolve()
> -                if not args.configuration.is_file():
> -                    raise ReleaseException(
> -                        f"Supplied configuration file is not found: {args.configuration}")
> -                    if args.debug:
> -                        configuration.logger.setLevel(logging.DEBUG)
> -                        for key, value in vars(args).items():
> -                            setattr(configuration, key, value)
>  
>          parser = argparse.ArgumentParser(description="Automatic releases bot", prog='release-bot')
>          parser.add_argument("-d", "--debug", help="turn on debugging output",
> @@ -62,4 +53,13 @@ class CLI:
>          parser_init.set_defaults(func=run_init)
>  
>          args = parser.parse_args()
> +        if args.debug:
> +            configuration.logger.setLevel(logging.DEBUG)
> +            for key, value in vars(args).items():
> +                setattr(configuration, key, value)
> +        if args.configuration:
> +            configuration.configuration = Path(args.configuration).resolve()
> +            if not configuration.configuration.is_file():
> +                raise ReleaseException(
> +                    f"Supplied configuration file is not found: {args.configuration}")
>          args.func(args)
> ```
> 
> I understand that argparse is hard to work with, feel free [to get inspired here](https://github.com/TomasTomecek/ansible-bender/blob/master/ansible_bender/cli.py#L321).

Thanks @TomasTomecek will work on it

Reply:
By patch I meant the diff in the big comment above.

Reply:
@TomasTomecek, I think this is better!